torch/jit/python_frontend.py

```diff
@@ -0,0 +1,285 @@
+import torch
+import sys
+import ast
+import inspect
+from functools import partial
+from collections import namedtuple
+from torch._C._jit_tree_views import *
+
+# TODO: improve error reporting (show source)
+#       once this is done, convert all asserts into checks with nicer error messages
+
+PY2 = sys.version_info[0] == 2
+_reserved_prefix = '__jit'
+
+
+def get_jit_ast(fn):
+    source = dedent(inspect.getsource(fn))
+    py_ast = ast.parse(source)
+    if len(py_ast.body) != 1 or not isinstance(py_ast.body[0], ast.FunctionDef):
+        raise RuntimeError("expected a single top-level function")
+    return build_def(SourceRangeFactory(source), py_ast.body[0])
+
+
+def dedent(source):
+    lines = source.split('\n')
+    indent_depth = 0
+    while lines[0][indent_depth] == ' ':
+        indent_depth += 1
+    return '\n'.join(l[indent_depth:] for l in lines)
+
+
+class Builder(object):
+    def __call__(self, ctx, node):
+        try:
+            method = getattr(self, 'build_' + node.__class__.__name__)
+        except AttributeError:
+            raise RuntimeError(node.__class__.__name__ + " isn't a supported Python feature")
```

Comment:
The stack trace for this error isn't going to tell you anything about which line caused it, as far as I can tell. Can all of these exceptions that get thrown include the source line of the user's program that prompted it?

Reply:
Yes, that's the plan. See my other comment about error handling. It's planned as a next PR. I wanted to get feedback on this/get this skeleton in, and then I can get to writing nice looking error messages